crates/storage/provider/src/providers/database/provider.rs

```diff
@@ -247,15 +251,15 @@ impl<TX: DbTx> DatabaseProvider<TX> {
     ///
     /// # Arguments
     /// * `segment` - The segment of the snapshot to check against.
-    /// * `block_number` - Requested block number
+    /// * `number` - Requested block or tx number
```

Comment:
is this basically an index key?
```suggestion
    /// * `index_key` - Requested index key, usually a block or transaction number.
```